src/phongo_ini.c

```diff
@@ -14,96 +14,82 @@
  * limitations under the License.
  */
 
-#include "bson/bson.h"
-#include "mongoc/mongoc.h"
+#include <stdio.h>
 
 #include <php.h>
 #include <main/php_open_temporary_file.h>
-#include <ext/date/php_date.h>
 
 #include "php_phongo.h"
 #include "phongo_ini.h"
+#include "phongo_log.h"
 
 ZEND_EXTERN_MODULE_GLOBALS(mongodb)
 
-static void phongo_log(mongoc_log_level_t log_level, const char* log_domain, const char* message, void* user_data)
+static FILE* phongo_ini_tmp_file(const char* tmp_dir)
 {
-	struct timeval tv;
-	time_t         t;
-	zend_long      tu;
-	zend_string*   dt;
+	time_t       t;
+	int          fd = -1;
+	char*        prefix;
+	int          len;
+	zend_string* filename = NULL;
+	FILE*        stream   = NULL;
 
-	bson_gettimeofday(&tv);
-	t  = tv.tv_sec;
-	tu = tv.tv_usec;
+	/* TODO: consider using time() retval directly instead of time_t struct */
+	time(&t);
+	len = spprintf(&prefix, 0, "PHONGO-%ld", t);
 
-	dt = php_format_date((char*) ZEND_STRL("Y-m-d\\TH:i:s"), t, 0);
+	/* TODO: Refactor this to use fdopen (see: PHPC-2181) */
```

Comment:
This was discovered while working on this PR in any earlier phase and I created [PHPC-2181](https://jira.mongodb.org/browse/PHPC-2181) to defer this task since it's a bit of a rabbit hole and we've been using `php_open_temporary_fd()` for a very long time already.